python/pyarrow/_csv.pyx

```diff
@@ -121,6 +121,59 @@ cdef class ReadOptions(_Weakrefable):
     encoding : str, optional (default 'utf8')
         The character encoding of the CSV data.  Columns that cannot
         decode using this encoding can still be read as Binary.
+
+    Example
+    -------
+
+    Defining an example file from bytes object:
+
+    >>> import io
+    >>> s = b'''1,2,3
+    ... Flamingo,2,2022-03-01
+    ... Horse,4,2022-03-02
+    ... Brittle stars,5,2022-03-03
+    ... Centipede,100,2022-03-04'''
```

Comment:
Checking the pandas guide, where we seem to solve this by defining it as a single-line string with manual `\n` in it for line breaks, but then also print it first to be able to see what it looks like. For example see https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#specifying-column-data-types

Reply:
In this case the docstring of a a single-line string is printed correctly interactively (in IPython or Jupyter Notebook at least) and the lines are duplicated by calling `print()` which makes the whole example longer:

```python
    Examples
    --------

    Defining an example file from bytes object:

    >>> import io
    >>> s = b'''1,2,3
Flamingo,2,2022-03-01
Horse,4,2022-03-02
Brittle stars,5,2022-03-03
Centipede,100,2022-03-04'''
    >>> print(s.decode())
    1,2,3
    Flamingo,2,2022-03-01
    Horse,4,2022-03-02
    Brittle stars,5,2022-03-03
    Centipede,100,2022-03-04
```

But if print is omitted, the example in html version is missing a better visual of the data. 

I do not have a good idea what I would suggest at this moment, but will think about it and add ideas here.

Reply:
I think if you do include the "print", I would use explicit newlines (`\n`) in `s`, so that this becomes a one-liner (and avoids making the example a lot longer). 
Or, if you keep the multi-line string as in your example above, I think you can also leave out the "print" call.

Reply:
Yes, I used `\n` in the example above and the docstrings got printed multi-lined in interactive session.

Corrected docstrings for `csv.read_csv()`:

```
    Examples
    --------

    Defining an example file from bytes object:

    >>> import io
    >>> s = b'animals,n_legs,entry\nFlamingo,2,2022-03-01\nHorse,4,2022-03-02\nBrittle stars,5,2022-03-03\nCentipede,100,2022-03-04
    >>> print(s.decode())

    Reading from the file
...
```

and docstring print for `csv.read_csv?`:

```
    Examples
    --------

    Defining an example file from bytes object:

    >>> import io
    >>> s = b'animals,n_legs,entry
Flamingo,2,2022-03-01
Horse,4,2022-03-02
Brittle stars,5,2022-03-03
Centipede,100,2022-03-04
    >>> print(s.decode())

    Reading from the file
...
```

Reply:
Ah, I see (didn't get that what you showed was the output from `?`). Hmm, for the raw docstring that indeed gives a duplication. But for the online / html rendered docs, it would be useful to include ..

Reply:
Idea from our call: use `\\n` if that also renders fine in the html docs?